### PR TITLE
Compatibility with ggplot2 3.5.0

### LINKED ISSUE #481: failure in basic ggpairs example with GGally 2.20/ggplot2 3.4.4.9000/r-devel
I'm guessing there is some incompatibility here that may be resolved with a version update, but haven't dug in yet ...

```
packageVersion("GGally"); packageVersion("ggplot2")
## [1] ‘2.2.0’
## [1] ‘3.4.4.9000’
data(flea)
ggpairs(flea, columns = 2:4)
```

> Error in as.character(call[[1]]) : 
  cannot coerce type 'closure' to vector of type 'character'

```
traceback()
11: startsWith(as.character(call[[1]]), "scale_")
10: scale_y_continuous()
9: (function (data, mapping, ..., rescale = FALSE) 
   {
       mapping <- mapping_color_to_fill(mapping)
       p <- ggplot(data, mapping) + scale_y_continuous()
       if (identical(rescale, TRUE)) {
           p <- p + stat_density(aes(y = after_stat(!!as.name("scaled")) * 
               diff(range(x, na.rm = TRUE)) + min(x, na.rm = TRUE)), 
               position = "identity", geom = "line", ...)
       }
       else {
           p <- p + geom_density(...)
       }
       p
   })(data = structure(list(species = structure(c(1L, 1L, 1L, 1L, 
   
{redacted}

   2L, 2L, 2L, 2L, 2L, 2L), levels = c("Concinna", "Heikert.", "Heptapot."
   ), class = "factor"), tars1 = c(191L, 

{redacted}

       74L, 86L, 92L, 78L, 66L, 55L, 84L, 86L, 68L, 86L, 89L, 88L
       )), class = "data.frame", row.names = c(NA, -74L)), mapping = structure(list(
       x = ~tars1), class = "uneval"))
8: do.call(original_fn, allParams)
7: fn(pm$data, plotObj$mapping)
6: getPlot(pm, i, j)
5: `[.ggmatrix`(pm, i, j)
4: pm[i, j]
3: ggmatrix_gtable(x, ...)
2: print.ggmatrix(x)
1: (function (x, ...) 
   UseMethod("print"))(x)
```

### PR DESCRIPTION
**Pull request by teunbrand**
TL;DR: This PR aims to fix #481.

We have been preparing a new release of ggplot2, and during a reverse dependency check, it became apparent that the prospective ggplot 3.5.0 would break GGally.
Given GGally's important role in the R ecosystem, we felt it was prudent to help out to transition to the new version.
The culprit is that ggplot2 changed the way legends are placed in the gtable, see https://github.com/tidyverse/ggplot2/pull/5488.
This PR handles the new legend structure, and should work with both the current and the development version of ggplot2.
To test the code changes with the release candidate, you can install it with the code below:

```r
remotes::install_github("tidyverse/ggplot2", ref = remotes::github_pull("5592"))
```

The release of ggplot2 3.5.0 is scheduled at the end of January / early Februari. The progress can be tracked in https://github.com/tidyverse/ggplot2/issues/5588.
We are hoping that this PR might help GGally be ready around that time to soften the impact on downstream packages.

### CODE REVIEW

**Comment by teunbrand** [comment]
This is just a quick little update to let you know that we've picked the 12th of February as release date.

**Comment by teunbrand** [comment]
And this is a kind reminder that the release is scheduled soon.

**Comment by schloerke** [comment]
@teunbrand Thank you for the wonderful PR and reminders!

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # GGally (development version)
 
+* Fix compatibility with ggplot2 3.5.0 (#481)
```
Comment (schloerke) [inline NEWS.md:3]:
```suggestion
* Fix compatibility with ggplot2 3.5.0 (@teunbrand, #481)
```
